util/chplenv/chpl_arch.py

```diff
@@ -588,29 +603,42 @@ def get(flag, map_to_compiler=False, get_lcd=False):
     compiler_val = chpl_compiler.get(flag)
     isprgenv = compiler_is_prgenv(compiler_val)
     if map_to_compiler and not isprgenv:
-        # Map flag/arch to compiler flag/argument
+        # Map arch to compiler argument
         # Don't do this for PrgEnv compiles since the compiler driver
         # handles specialization.
         version = get_compiler_version(compiler_val)
-        (flag, arch) = argument_map.find(arch, compiler_val, version)
-    elif arch and arch != 'none' and arch != 'unknown':
-        if is_known_arm(arch):
-            flag = 'cpu'
-        else:
+        arch = argument_map.find(arch, compiler_val, version)
+    if arch and arch != 'none' and arch != 'unknown':
+        native_arch = get_native_machine()
+        # x86 uses -march= where the others use -mcpu=
+        is_x86_32 = native_arch.startswith('i') and native_arch.endswith('86')
+        if (native_arch in ['amd64', 'x86_64']) or is_x86_32:
```

Comment:
Should this also check for x86 (This is the Microsoft term for 32-bit x86), x86-64 (This is a spelling variant), and x64 (which is a Microsoft term for x86-64)?

Can this code (and the 32-bit check) go into a separate function that computes "Is it an x86 variant"?